app/Http/Controllers/Api/v1/EntityApiController.php

```diff
@@ -41,6 +41,27 @@ public function index(Campaign $campaign)
             ->appends(request()->except(['page', 'lastSync'])));
     }
 
+    /**
+     * @return \Illuminate\Http\Resources\Json\AnonymousResourceCollection
+     * @throws \Illuminate\Auth\Access\AuthorizationException
+     */
+    public function recent(Campaign $campaign)
```

Comment:
Please make this a new `RecentEntityApiController` controller., this one is already big enough :)